### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by risantos**
- Updates comments to `JSDoc`, adding some descriptions and declarations.
- Sets some type definitions with `@typedef`, so that the right types are inherited by those using the module.
- Formats `Readme.md`.

### CODE REVIEW

File: lib/mixins/health-check.js [hunk 1]
```diff
@@ -3,8 +3,11 @@
 /**
  * Health check handling mixin.
  *
- * @param {Reader|Writer} client
- * @api private
+ * @typedef {import('../reader')} Reader
+ * @typedef {import('../writer')} Writer
+ * @param {Reader|Writer} client - Reader or Writer instance.
```
Comment (Americas) [inline lib/mixins/health-check.js]:
```suggestion
 * @param {Reader | Writer} client - Reader or Writer instance.
```

File: lib/mixins/close.js [hunk 2]
```diff
@@ -2,26 +2,37 @@
...
+ * @private
  */
 
 module.exports = reader => {
   let pending = 0;
 
+  /**
+   * Decrement pending counter.
+   * @function decr
+   */
```
Comment (Americas) [inline lib/mixins/close.js:26]: Should there be a new line after these comments?

File: lib/mixins/ready.js [hunk 3]
```diff
@@ -9,8 +9,10 @@ const debug = require('debug')('nsq:ready');
 /**
  * RDY handling mixin.
  *
- * @param {Connection} conn
- * @api private
+ * @typedef {import('../connection')} Connection
```
Comment (Americas) [inline lib/mixins/ready.js]: Move up?

File: lib/mixins/close.js (hunk 2 above)
Comment (risantos) [reply lib/mixins/close.js:26]: I'll adjust all to have a line between comment blocks and declaration for consistency. (except for single-line comments)

**Review by Americas — APPROVED** [review]

**Pull request merged**